nomos-blend/core/src/conn_maintenance.rs

```diff
@@ -125,6 +125,9 @@ where
     pub fn reset(&mut self) -> (HashSet<Multiaddr>, HashSet<Multiaddr>) {
         let (malicious_peers, unhealthy_peers) = self.analyze_monitors();
 
+        // Maintain the list of malicious peers
+        self.malicious_peers.extend(malicious_peers.iter().cloned());
```

Comment:
why cloned? you can just `malicious_peers.into_iter()` to take ownership.

Reply:
It's because we have to return those malicious peers as a `peers_to_close` from this function. In other words, the `malicious_peers` is used for `peers_to_close` and `self.malicious_peers`.